src/main/java/com/pinterest/secor/uploader/FormatConvertingUploadManager.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.secor.uploader;
+
+import com.pinterest.secor.common.LogFilePath;
+import com.pinterest.secor.common.SecorConfig;
+import com.pinterest.secor.io.FileReader;
+import com.pinterest.secor.io.FileWriter;
+import com.pinterest.secor.io.KeyValue;
+import com.pinterest.secor.uploader.UploadManager;
+import com.pinterest.secor.util.CompressionUtil;
+import com.pinterest.secor.util.FileUtil;
+import com.pinterest.secor.util.IdUtil;
+import com.pinterest.secor.util.ReflectionUtil;
+
+import java.io.File;
+
+import org.apache.hadoop.io.compress.CompressionCodec;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages uploads using the configured upload manager
+ * Converts the local files to the specified format before upload
+ *
+ * @author Jason Butterfield (jason.butterfield@outreach.io)
+ */
+public class FormatConvertingUploadManager extends UploadManager {
+    private static final Logger LOG = LoggerFactory.getLogger(FormatConvertingUploadManager.class);
+
+    private UploadManager mUploadManager;
```

Comment:
Hungarian notation has been pretty much discarded thanks to modern IDEs and patterns requiring minimizing number of variables in the first place.  If you can't keep track in your head which ones are private and which are public, you have too many member variables and should be breaking your class up.  If you can keep track in your head, you don't need an indication of it being a member variable.

Reply:
I don't disagree, but this whole repo uses this notation style and I think consistency is nice.